sentry-core/src/main/java/io/sentry/core/SentryCommonOptions.java

```diff
@@ -0,0 +1,263 @@
+   */
+  public void applyTo(SentryOptions options) {
+    if (dsn != null) {
+      options.setDsn(dsn);
+    }
+    if (maxBreadcrumbs != null) {
+      options.setMaxBreadcrumbs(maxBreadcrumbs);
+    }
+    if (environment != null) {
+      options.setEnvironment(environment);
+    }
+    if (shutdownTimeoutMillis != null) {
+      options.setShutdownTimeout(shutdownTimeoutMillis);
+    }
+    if (flushTimeoutMillis != null) {
+      options.setFlushTimeoutMillis(flushTimeoutMillis);
+    }
+    if (readTimeoutMillis != null) {
+      options.setReadTimeoutMillis(readTimeoutMillis);
+    }
+    if (sampleRate != null) {
+      options.setSampleRate(sampleRate);
+    }
+    if (bypassSecurity != null) {
+      options.setBypassSecurity(bypassSecurity);
+    }
+    if (debug != null) {
+      options.setDebug(debug);
+    }
+    if (attachThreads != null) {
+      options.setAttachThreads(attachThreads);
+    }
+    if (attachStacktrace != null) {
+      options.setAttachStacktrace(attachStacktrace);
+    }
+    if (diagnosticLevel != null) {
+      options.setDiagnosticLevel(diagnosticLevel);
+    }
+    if (dist != null) {
+      options.setDist(dist);
+    }
+    if (release != null) {
+      options.setRelease(release);
+    }
+    if (sampleRate != null) {
+      options.setSampleRate(sampleRate);
+    }
+    if (serverName != null) {
+      options.setServerName(serverName);
+    }
+    if (inAppExcludes != null) {
+      for (String inAppExclude : inAppExcludes) {
+        options.addInAppExclude(inAppExclude);
+      }
+    }
+    if (inAppIncludes != null) {
+      for (String inAppInclude : inAppIncludes) {
+        options.addInAppInclude(inAppInclude);
+      }
+    }
+  }
+
+  public String getDsn() {
+    return dsn;
+  }
+
+  public void setDsn(String dsn) {
+    this.dsn = dsn;
+  }
+
+  public Long getShutdownTimeoutMillis() {
+    return shutdownTimeoutMillis;
+  }
+
+  public void setShutdownTimeoutMillis(Long shutdownTimeoutMillis) {
+    this.shutdownTimeoutMillis = shutdownTimeoutMillis;
+  }
+
+  public Long getFlushTimeoutMillis() {
+    return flushTimeoutMillis;
+  }
+
+  public void setFlushTimeoutMillis(Long flushTimeoutMillis) {
+    this.flushTimeoutMillis = flushTimeoutMillis;
+  }
+
+  public Integer getReadTimeoutMillis() {
+    return readTimeoutMillis;
+  }
+
+  public void setReadTimeoutMillis(Integer readTimeoutMillis) {
+    this.readTimeoutMillis = readTimeoutMillis;
+  }
+
+  public Boolean getBypassSecurity() {
+    return bypassSecurity;
+  }
+
+  public void setBypassSecurity(Boolean bypassSecurity) {
+    this.bypassSecurity = bypassSecurity;
+  }
+
+  public Boolean getDebug() {
+    return debug;
+  }
+
+  public void setDebug(Boolean debug) {
+    this.debug = debug;
+  }
+
+  public SentryLevel getDiagnosticLevel() {
+    return diagnosticLevel;
+  }
+
+  public void setDiagnosticLevel(SentryLevel diagnosticLevel) {
+    this.diagnosticLevel = diagnosticLevel;
```

Comment:
if you look at `SentryOptions.setDiagnosticLevel` you see that we don't allow null values, and here it is possible, if one sets it to null, it'll crash the app.

Reply:
In `applyTo` there is a check for null. The idea is that we apply only properties explicitly set by user.